### PR TITLE
Added CartService

### PR DESCRIPTION
**Pull request by jamesshenry**
Mostly boilerplate atm. Not sure where I should be getting the cart Guid from

### CODE REVIEW

**Comment by pobiega** [comment]
If you call the "add to cart" endpoint without a cart ID, it will create a new cart and give you the ID. You then reuse that ID when adding more things and to check out.

**Comment by jamesshenry** [comment]
Pretty sure it's working as intended now, but calling the endpoint with no guid causes a null reference exception, so perhaps the get cart api endpoint isn't implemented fully yet?

**Comment by pobiega** [comment]
> calling the endpoint with no guid causes a null reference exception

What endpoint are you calling? `GetCart` doesn't generate a new cart, `AddToCart` does :)

File: src/Congo.RazorPages/Services/ICartService.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Threading.Tasks;
+using Congo.Contracts.Responses.Cart;
+
+namespace Congo.RazorPages.Services
+{
+    public interface ICartService
+    {
+
+        Task<CartResponse> GetCart(Guid? cartId = null);
```
Comment (valincius) [inline src/Congo.RazorPages/Services/ICartService.cs:10]: Should not be nullable

File: src/Congo.RazorPages/Services/CartService.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public class CartService : ICartService
+    {
+        private readonly ICongoUserClient _client;
+
+        public CartService(ICongoUserClient client)
+        {
+            _client = client;
+        }
+        public async Task<CartResponse> GetCart(Guid? cartId)
```
Comment (valincius) [inline src/Congo.RazorPages/Services/CartService.cs:17]: Should not be nullable

File: src/Congo.RazorPages/Pages/Cart.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        }
+        public async Task<IActionResult> OnGetAsync()
+        {
+            string cartId;
+            if (Request.Cookies.ContainsKey("cartId"))
+                Cart = await _cartService.GetCart(Guid.Parse(Request.Cookies["cartId"]));
+            else
+            {
+                cartId = (await _cartService.GetCart()).ToString();
+                Response.Cookies.Append("cartId", cartId);
```
Comment (valincius) [inline src/Congo.RazorPages/Pages/Cart.cshtml.cs:27]: This isn't right, `GetCart` returns a `CartResponse` - if a `cartId` doesn't exist in the cookie then you should show a message like 'No items' or something.

You're sorta on the right track, we want this sort of logic for add to cart, but not here.

File: src/Congo.RazorPages/Pages/Cart.cshtml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿@page
+@model CartModel
+@{
+    ViewData["Title"] = "Cart";
+}
+
+<h1>Cart</h1>
+<div class="container-fluid">
+    <div class="row">
+        @foreach (var cartItem in Model.Cart.CartItems)
```
Comment (valincius) [inline src/Congo.RazorPages/Pages/Cart.cshtml:10]: Show a message if we have no items in our cart - or if we don't have a cart

File: src/Congo.Contracts/Clients/ICongoUserClient.cs [hunk 5]
```diff
@@ -14,5 +15,8 @@ public interface ICongoUserClient
 
         [Post("/api/products/{id}/purchase")]
         Task<ApiResponse<OrderConfirmationResponse>> PurchaseProduct(Guid id);
+
+        [Get("/api/cart/{id}")]
+        Task<ApiResponse<CartResponse>> GetCart(Guid? id);
```
Comment (valincius) [inline src/Congo.Contracts/Clients/ICongoUserClient.cs:20]: Id should not be nullable

File: src/Congo.RazorPages/Pages/Cart.cshtml.cs (hunk 3 above)
Comment (valincius) [reply src/Congo.RazorPages/Pages/Cart.cshtml.cs:27]: Need to catch error if we somehow have an invalid cart id

**Review by valincius — CHANGES_REQUESTED** [review]

**Comment by valincius** [comment]
Please clone this repo and make branches off of that in the future, forking makes it a little hard to check out and review your changes

**Comment by jamesshenry** [comment]
Will close this and start again

**Pull request closed without merging**